vql/darwin/xattr.go

```diff
@@ -0,0 +1,97 @@
+//go:build !windows
+// +build !windows
+
+package darwin
+
+import (
+	"context"
+
+	"github.com/Velocidex/ordereddict"
+	"github.com/ivaxer/go-xattr"
+	"www.velocidex.com/golang/velociraptor/accessors"
+	"www.velocidex.com/golang/velociraptor/acls"
+	vql_subsystem "www.velocidex.com/golang/velociraptor/vql"
+	"www.velocidex.com/golang/vfilter"
+	"www.velocidex.com/golang/vfilter/arg_parser"
+)
+
+type XAttrArgs struct {
+	Filename   *accessors.OSPath `vfilter:"required,field=filename,doc=Filename to inspect."`
+	Attributes []string          `vfilter:"optional,field=attribute,doc=Attribute to collect."`
+	Accessor   string            `vfilter:"optional,field=accessor,doc=File accessor"`
+}
+
+type XAttrFunction struct {
+	Filename   string
```

Comment:
No need to keep struct members. It's better to not keep state in vql functions if we can help it because it's hard to know when new instances are created